src/pages/utilities/FileSearch.tsx

```diff
@@ -0,0 +1,446 @@
+  const [files, fileCount] = useFlattenListResult<FileType>(fileResults);
+
+  const {
+    handleRowSelect,
+    rowSelection,
+    selectedRows,
+    setRowSelection,
+  } = useRowSelection<FileType>(files);
+
+  const [viewIndex, setViewIndex] = useState(0);
+
+  const onNextView = useCallback(() => {
+    setViewIndex((prev) => {
+      if (prev + 1 >= selectedRows.length) return 0;
+
+      return prev + 1;
+    });
+  }, [selectedRows]);
+
+  const onPrevView = useCallback(() => {
+    setViewIndex((prev) => {
+      if (prev - 1 < 0) return selectedRows.length - 1;
+
+      return prev - 1;
+    });
+  }, [selectedRows]);
+
+  const selectedId = useMemo(() => selectedRows[viewIndex]?.ID, [selectedRows, viewIndex]);
+
+  const columns = useMemo<UtilityHeaderType<FileType>[]>(
+    () => staticColumns,
+    [],
+  );
+
+  const renderTotalFiles = () => (
+    <div className="flex font-semibold">
+      <p className="text-panel-text-important">{fileCount ?? 0}</p>
+      &nbsp;
+      {fileCount <= 1 ? 'File' : 'Files'}
+    </div>
+  );
+
+  const renderMetadata = useMemo(() => {
+    const selectedSeriesCount = Object.keys(
+      groupBy(selectedRows.flatMap(x => x.SeriesIDs).map(x => x?.SeriesID), x => x?.ID),
+    ).length;
+    const selectedSize = prettyBytes(selectedRows.reduce((prev, cur) => prev + cur.Size, 0), { binary: true });
+    const selectedEpisodeCount = selectedRows.flatMap(x => x.SeriesIDs).map(x => x?.EpisodeIDs).length;
+
+    return (
+      <>
+        <div className="flex justify-between capitalize">
+          <span className="font-semibold">Series Count</span>
+          <span>{selectedSeriesCount}</span>
+        </div>
+        <div className="flex justify-between capitalize">
+          <span className="font-semibold">Episode Count</span>
+          <span>{selectedEpisodeCount}</span>
+        </div>
+        <div className="flex justify-between capitalize">
+          <span className="font-semibold">Total Size</span>
+          <span>{selectedSize}</span>
+        </div>
+      </>
+    );
+  }, [selectedRows]);
+
+  return (
+    <div className="flex grow flex-col gap-y-8">
+      <ShokoPanel title="File Search" options={renderTotalFiles()}>
+        <div className="flex items-center gap-x-3">
+          <Input
+            type="text"
+            placeholder="Search..."
+            startIcon={mdiMagnify}
+            id="search"
+            onChange={({ target: { value } }) => {
+              setSearch(_ => value);
+            }}
+            value={search}
+            inputClassName="px-4 py-3"
+          />
+          <Menu
+            selectedRows={selectedRows}
+            setSelectedRows={setRowSelection}
+          />
+        </div>
+      </ShokoPanel>
+      <div className="flex grow justify-between overflow-y-auto">
+        <div className="flex w-full rounded-md border border-panel-border bg-panel-background p-8 md:max-w-[65%] lg:max-w-[75%]">
+          {(fileCount ?? 0) > 0
```

Comment:
```suggestion
          {fileCount > 0
```
`fileCount` is always defined